front/appliance.form.php

```diff
@@ -0,0 +1,104 @@
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2020 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+use Glpi\Event;
+
+include ('../inc/includes.php');
+
+Session::checkRight('appliance', READ);
+
+if (empty($_GET["id"])) {
+   $_GET["id"] = "";
+}
+if (!isset($_GET["withtemplate"])) {
+   $_GET["withtemplate"] = "";
+}
+
+$app = new Appliance();
+
+if (isset($_POST["add"])) {
+   $app->check(-1, CREATE, $_POST);
+
+   if ($newID = $app->add($_POST)) {
+      Event::log($newID, "appliance", 4, "inventory",
+                 sprintf(__('%1$s adds the item %2$s'), $_SESSION["glpiname"], $_POST["name"]));
+      if ($_SESSION['glpibackcreated']) {
+         Html::redirect($app->getLinkURL());
+      }
+   }
+   Html::back();
+
+} else if (isset($_POST["delete"])) {
+   $app->check($_POST["id"], DELETE);
+   $app->delete($_POST);
+
+   Event::log($_POST["id"], "appliance", 4, "inventory",
+              //TRANS: %s is the user login
+              sprintf(__('%s deletes an item'), $_SESSION["glpiname"]));
+   $app->redirectToList();
+
+} else if (isset($_POST["restore"])) {
+   $app->check($_POST["id"], DELETE);
+
+   $app->restore($_POST);
+   Event::log($_POST["id"], "appliance", 4, "inventory",
+              //TRANS: %s is the user login
+              sprintf(__('%s restores an item'), $_SESSION["glpiname"]));
+   $app->redirectToList();
+
+} else if (isset($_POST["purge"])) {
+   $app->check($_POST["id"], PURGE);
+
+   $app->delete($_POST, 1);
+   Event::log($_POST["id"], "appliance", 4, "inventory",
+              //TRANS: %s is the user login
+              sprintf(__('%s purges an item'), $_SESSION["glpiname"]));
+   $app->redirectToList();
+
+} else if (isset($_POST["update"])) {
+   $app->check($_POST["id"], UPDATE);
+
+   $app->update($_POST);
+   Event::log($_POST["id"], "appliance", 4, "inventory",
+              //TRANS: %s is the user login
+              sprintf(__('%s updates an item'), $_SESSION["glpiname"]));
+   Html::back();
+
```

Comment:
Missing dellieu and addlieu

Reply:
Those are specific to the plugin; and seems related to ApplianceRelation problem described in PR description.

Reply:
we can't use the Appliances without the relations (explanation done by mail).

Reply:
Well, what your are talking about is to setup a CommonDBRelation on two other CommonDBRelation. As said, I'm not sure it can work; and it will certainly be a nightmare from an UI point of view.
We'll make tests and work on that specific point on another branch; but it certainly won't be finished for the 9.5.2 release.